### PR TITLE
feat: gemini compatibility and skill path fixes

### PR DESCRIPTION
**Pull request by N4M3Z**
This PR improves Gemini CLI compatibility by adding missing tool mappings and fixing a regression where skill subdirectories were lost during the assembly process.

### CODE REVIEW

File: templates/init/.pre-commit-config.yaml [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
+            language: system
+            pass_filenames: false
+
+          - id: semgrep
+            name: semgrep OWASP
+            entry: semgrep scan --config=p/owasp-top-ten --metrics=off --quiet .
+            language: system
+            pass_filenames: false
+
+          - id: validate-adr
```
Comment (l3hox) [inline templates/init/.pre-commit-config.yaml]: **[BLOCKER — Ops]** This hook references `scripts/validate-adr.py --test`, but `templates/init/scripts/` does not exist on this branch. Every module created by `forge init` will fail `git commit` immediately with a file-not-found error. Either ship `templates/init/scripts/validate-adr.py` (and any dependencies) or drop this hook from the template until the script lands.

File: src/transform/mod.rs [hunk 2]
```diff
@@ -33,6 +34,17 @@ pub fn apply_rules(
                 let (stem, extension) = split_extension(&current_filename);
                 let kebab = to_kebab_case(&stem);
                 current_filename = format!("{kebab}{extension}");
+                current_content =
+                    crate::assemble::map_field(&current_content, "name", to_kebab_case);
```
Comment (l3hox) [inline src/transform/mod.rs:38]: **[BLOCKER — Dev/QA/Docs]** `KebabCase` now mutates `name:` frontmatter unconditionally — no `kind` guard, unlike `KebabCaseAgents` below which guards on `kind == "agents"`. Because `name` is in `keep_fields` for agents and skills, this mutation lands in deployed output.

A skill with `name: MySkill` in frontmatter exits assembly as `name: my-skill`, silently. This is semantic corruption of an identity field.

Pick one fix:
- **Add the `kind == "agents"` guard here** (making `KebabCase` and `KebabCaseAgents` differ only in filename scope), or
- **Remove the `map_field` call** (making `KebabCase` strictly filename-only, and `KebabCaseAgents` the only variant that touches frontmatter).

Either way, add a test: `apply_rules_kebab_case_preserves_name_field_for_non_agent_kinds` — it should fail against the current code.

File: src/assemble/frontmatter.rs [hunk 3]
```diff
@@ -58,6 +58,52 @@ pub fn strip_frontmatter(content: &str, keep_fields: &[&str]) -> String {
...
+                let after_colon = &line[colon_pos + 1..];
+                let value = after_colon.trim();
+
+                // Find existing spacing after colon to preserve it
+                let spacing = &after_colon[..after_colon.len() - after_colon.trim_start().len()];
+                let spacing = if spacing.is_empty() { " " } else { spacing };
+
+                new_yaml.push_str(&line[..=colon_pos]);
+                new_yaml.push_str(spacing);
+                new_yaml.push_str(&mapper(value));
```
Comment (l3hox) [inline src/assemble/frontmatter.rs]: **[BLOCKER — Dev/QA]** `value` is passed to `mapper` including any surrounding quote characters. A frontmatter field like `name: "SecurityArchitect"` gives the mapper the input `"SecurityArchitect"` (with quotes), and `to_kebab_case` will happily emit `"security-architect"` — quotes now part of the identifier, corrupting downstream YAML consumers.

Related QA findings against this function:
- **Inline comments dropped**: `name: X  # comment` survives the first-colon match, but `trim()` collapses everything into the mapper input; the trailing `# comment` is silently lost.
- **Block scalars corrupted**: `name: |` plus indented continuation lines — only the header line is matched; continuation lines are emitted verbatim, no longer associated with the replaced key.

Line-by-line parsing cannot sustain these cases. Recommendation: round-trip through `serde_yaml` for surgical field updates. Short-term patch: strip surrounding `"`/`'` before calling the mapper and re-add on output if originally quoted.

Add test: `map_field_preserves_quoted_value` (should fail today).

File: src/cli/assemble/pipeline.rs [hunk 4]
```diff
@@ -27,8 +27,9 @@ pub fn assemble_source(
     module_root: &Path,
     provider_name: &str,
     keep_fields: &[String],
-    tool_mappings: &HashMap<String, String>,
-    model_tiers: &HashMap<String, String>,
+    _tool_mappings: &HashMap<String, String>,
```
Comment (l3hox) [inline src/cli/assemble/pipeline.rs]: **[Dev]** `_tool_mappings` is now a dead parameter — tool remapping moved to `transform::apply_rules` via the `RemapTools` rule. Underscore-prefixing silences the warning but leaves a misleading signature. Remove the parameter and clean up the single caller in `src/cli/assemble/mod.rs`.

File: src/provider/mod.rs [hunk 5]
```diff
@@ -56,7 +58,7 @@ pub struct ProviderConfig {
     pub assembly: Option<Vec<String>>,
     pub deploy: Option<Vec<String>>,
     pub keep_fields: Option<HashMap<String, Vec<String>>>,
-    pub models: Option<HashMap<String, String>>,
+    pub models: Option<HashMap<String, Vec<String>>>,
```
Comment (l3hox) [inline src/provider/mod.rs:61]: **[Dev]** Type changed from `HashMap<String, String>` to `HashMap<String, Vec<String>>`, but the only consumer (`map_model_tier` in `src/cli/assemble/pipeline.rs`) uses `.first()` and discards the rest. If the intent is "primary model with fallbacks," the fallback logic is unimplemented and the `Vec` is advertising a contract the code doesn't fulfill. Either implement fallback traversal or document explicitly why it's a list (e.g. qualifier matching). Not a blocker, but the contract should match the code.

File: src/assemble/pipeline.rs [hunk 6]
```diff
@@ -132,11 +132,19 @@ pub fn assemble_module(
...
+    if kind.is_empty() {
+        return relative_path;
+    }
+
     relative_path
-        .rsplit_once('/')
-        .map_or(relative_path, |(_, name)| name)
+        .strip_prefix(kind)
+        .and_then(|p| p.strip_prefix('/'))
+        .unwrap_or(relative_path)
```
Comment (l3hox) [inline src/assemble/pipeline.rs:147]: **[Dev]** `unwrap_or(relative_path)` silently returns the full unstripped path if `kind` is non-empty but the prefix doesn't match — a regression from the old `rsplit_once('/')` fallback which at least gave the bare filename. Unlikely to trigger in practice, but worth a `debug_assert!` or a warning log on the fallback branch so an unexpected source layout surfaces instead of producing a malformed output path.

File: templates/init/.pre-commit-config.yaml [hunk 7]
```diff
@@ -1,11 +1,46 @@
...
+          - id: ruff
+            name: ruff check
+            entry: ruff check .
+            language: system
+            pass_filenames: false
+            types: [python]
+
+          - id: gitleaks
+            name: gitleaks
+            entry: gitleaks detect --no-banner --no-git -s .
```
Comment (l3hox) [inline templates/init/.pre-commit-config.yaml]: **[Ops]** `--no-git` scans only the working tree. Secrets committed in past commits but absent from the current worktree will not be caught — meaningful scope gap for a secret-scanning gate. Consider dropping `--no-git` (or documenting why it's intentional, e.g. local-only pre-commit context).

File: templates/init/.pre-commit-config.yaml [hunk 8]
```diff
@@ -1,11 +1,46 @@
...
+            language: system
+            pass_filenames: false
+
+          - id: semgrep
+            name: semgrep OWASP
+            entry: semgrep scan --config=p/owasp-top-ten --metrics=off --quiet .
```
Comment (l3hox) [inline templates/init/.pre-commit-config.yaml:33]: **[Ops]** `--config=p/owasp-top-ten` pulls a remote ruleset from semgrep.dev on every run with no version pinning. Ruleset updates can silently change results between commits. Pin to a specific ruleset commit hash, or vendor the config into the scaffold. Separately: this hook and `gitleaks` both run full-tree with `pass_filenames: false` on every commit — consider `stages: [pre-push]` for the full-tree scanners to keep `pre-commit` fast.

Related: forge-cli's own `.github/workflows/quality.yaml` installs gitleaks/semgrep directly AND runs prek, executing both scanners twice per PR. Worth consolidating on prek as the single orchestrator.

File: README.md [hunk 9]
```diff
@@ -82,16 +82,18 @@ When a file exists in both `user/` and root, `user/` wins. Files only in `user/`
...
 providers:
     claude:
         target: ".claude"
     gemini:
         target: ".gemini"
+        aliases:
+            - geminicli
         assembly:
-            - kebab-case
+            - kebab-case-agents
```
Comment (l3hox) [inline README.md:96]: **[Docs]** The example now shows `kebab-case-agents` but there's no explanation anywhere user-facing of what it does (agent filenames only, not skills/rules) or how it differs from `kebab-case`. Add a one-line scope note under this block, e.g.:

> `kebab-case-agents` rewrites agent filenames and their `name:` frontmatter to kebab case; skills and rules are preserved as-is. Use `kebab-case` if you want the same transform applied to all content kinds.

Also update `CLAUDE.md:56` (assembly rules list) which still references only `kebab-case`.

File: GEMINI.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+When deploying to Gemini (`.gemini/`), `forge-cli` automatically transforms content to match Gemini CLI conventions:
+
+### 1. Tool Name Remapping
+`PascalCase` tool names in backticks are remapped to their `snake_case` equivalents using `config/remap-tools.yaml`.
+- `` `Read` `` becomes `` `read_file` ``
+- `` `Bash` `` becomes `` `run_shell_command` ``
+- `` `Complete` `` becomes `` `complete_task` ``
+
+### 2. Nested Pathing
+- **Agents:** Filenames are converted to kebab-case (e.g., `agents/SecurityArchitect.md` → `agents/security-architect.md`).
```
Comment (l3hox) [inline GEMINI.md]: **[Docs]** "Agents: Filenames are converted to kebab-case" is incomplete — it doesn't mention that the `name:` frontmatter field is also rewritten, and it doesn't distinguish the two rules (`kebab-case` vs `kebab-case-agents`). Since `GEMINI.md` is the primary reference for Gemini-side consumers, an inaccurate transformation guide here propagates wrong mental models. Clarify that: (1) only the `kebab-case-agents` rule runs by default for Gemini, (2) both the filename and the `name:` field are transformed, (3) skills and rules retain their PascalCase filenames and nested paths.

**Review by l3hox — COMMENTED** [review]
Developer council review (4 specialists: Dev, QA, Docs, Ops; 3 rounds). Inline comments below capture each recommendation on the exact line.

**Merge blockers** (2 required, 1 strongly recommended):
1. `templates/init/.pre-commit-config.yaml` references `scripts/validate-adr.py` but the script is not in the template tree — every scaffolded module breaks on first commit.
2. `src/transform/mod.rs` `KebabCase` arm now mutates `name:` frontmatter unconditionally — unguarded by `kind`, silently corrupts identity field.
3. `src/assemble/frontmatter.rs::map_field` strips surrounding quotes from values and drops inline comments (line-based parsing can't handle YAML edge cases).

**Follow-ups** noted inline: dead `_tool_mappings` param, unused `Vec<String>` fallback contract on `models`, silent-fallback in `extract_relative_within_kind`, README/GEMINI.md scope notes for `kebab-case` vs `kebab-case-agents`, `gitleaks --no-git` scope, unpinned `semgrep` remote ruleset, CI double-execution of gitleaks/semgrep (prek + direct install).

All 367 tests pass today. Recommended new tests: non-agent `KebabCase` preserves `name:` (would fail today — catches bug 2); `map_field` round-trips quoted values (would fail today — catches bug 3); `forge init` scaffold path-existence check (would catch bug 1).

**Comment by N4M3Z** [comment]
Addressed all findings in 778de72 → latest push. Summary:

**Blocker 1** (validate-adr template) — Removed. forge-core specific, not a universal module concern.

**Blocker 2** (KebabCase name mutation) — Intentional. `KebabCase` transforms both filename and `name:` for all kinds. `KebabCaseAgents` exists for the agents-only variant. No change needed.

**Blocker 3** (map_field quoted values) — Fixed. Replaced line-based parsing with `serde_yaml` round-trip. Handles quoted values, block scalars, and edge cases correctly. Added tests for double-quoted and single-quoted values.

**Dead `_tool_mappings`** — Removed parameter and cleaned up caller.

**gitleaks `--no-git`** — Dropped. Template now scans git history.

**semgrep/gitleaks stages** — Moved to `stages: [pre-push]` for faster pre-commit.

**README/GEMINI docs** — Added assembly transforms table to README. Clarified `kebab-case-agents` scope in GEMINI.md.

**Vec<String> contract / silent fallback** — Noted for future follow-up.
